clients/drcachesim/tools/view.cpp

```diff
@@ -95,15 +95,34 @@ view_t::initialize()
 bool
 view_t::process_memref(const memref_t &memref)
 {
-    if (!type_is_instr(memref.instr.type) &&
-        memref.data.type != TRACE_TYPE_INSTR_NO_FETCH)
+    if (instr_count < knob_skip_refs || instr_count >= (knob_skip_refs + knob_sim_refs)) {
+        if (type_is_instr(memref.instr.type) ||
+            memref.data.type == TRACE_TYPE_INSTR_NO_FETCH)
+            ++instr_count;
         return true;
+    }
 
-    if (instr_count < knob_skip_refs || instr_count >= (knob_skip_refs + knob_sim_refs)) {
-        ++instr_count;
+    if (memref.marker.type == TRACE_TYPE_MARKER) {
+        if (memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP)
```

Comment:
nit: Consider using a switch-case statement on the marker_type here. I think it will improve readability here as well as automated checks such as clang-tidy will complain about missing cases [1].

[1] https://clang.llvm.org/extra/clang-tidy/checks/hicpp-multiway-paths-covered.html#hicpp-multiway-paths-covered

Reply:
Done.